include/RevCPU.h

```diff
@@ -212,6 +221,63 @@ class RevCPU : public SST::Component{
   RevLoader *Loader;                  ///< RevCPU: RISC-V loader
   std::vector<RevProc *> Procs;       ///< RevCPU: RISC-V processor objects
   bool *Enabled;                      ///< RevCPU: Completion structure
+      
+  // Global table of threads
+  // - The actual RevThread objects live in this table 
+  // - Only place where the presence of a RevThread on this CPU is guanranteed throughout execution
+  std::map<uint32_t, std::shared_ptr<RevThread>> Threads;
+
+  // 2D vector to hold threads assigned to each processor and its corresponding harts.
+  // AssignedThreads[i][j] holds the RevThread executing on processor 'i' and hart 'j'.
+  // Oversubscription is not supported in hardware 
+  // (ie. AssignedThreads.at(i).size() will never exceed the number of HARTS)
+  std::vector<std::vector<std::shared_ptr<RevThread>>> AssignedThreads;
+
+  // Initializes a RevThread object.
+  // - Moves it to the 'Threads' map 
+  // - Adds it's ThreadID to the ThreadQueue to be scheduled
+  void InitThread(std::shared_ptr<RevThread> ThreadToInit);
```

Comment:
Should be
```c++
void InitThread(const std::shared_ptr<RevThread>& ThreadToInit);
```

Only pass `std::shared_ptr` by value when assigning ownership to a new owner, or when multiple threads (on the host) are using the same `std::shared_ptr` and you could have race conditions. Otherwise each call and return incurs an atomic increment/decrement.